### PR TITLE
Ae 1446 componente resultado questionário

### PR DESCRIPTION
**Pull request by KaynanWallen**
## Summary by CodeRabbit

* **New Features**
  * Introduced comprehensive quiz result displays, including headers, performance metrics, and subject-level statistics.
  * Added result mode to quiz alternatives, showing correct and incorrect answers with visual indicators.
  * Enhanced quiz result pages with detailed summaries and progress bars for overall and subject-specific performance.
  * Added interactive subject list with clickable cards to explore quiz results by subject.
  * Provided detailed question-level results filtered by subject with clickable questions for deeper review.
  * Standardized quiz state naming and unified answer tracking within quiz questions for improved consistency.
  * Added multiple new storybook examples demonstrating quiz result UI components and variants.

* **Tests**
  * Expanded test coverage for all new quiz result components and scenarios, ensuring accurate rendering and interaction handling.
  * Improved quiz state management tests to cover all quiz types and edge cases, verifying consistent answer tracking and progress calculation.
  * Added detailed tests for quiz navigation, finishing flows, and timer lifecycle management.
  * Added comprehensive tests for question answer status, result components, and user interaction callbacks.

### CODE REVIEW

File: src/components/Quiz/Quiz.tsx [hunk 1]
```diff
@@ -117,15 +146,34 @@ const QuizContent = forwardRef<
...
-  const alternatives = currentQuestion?.options?.map((option) => ({
-    label: option.option,
-    value: option.id,
-  }));
+  const alternatives = currentQuestion?.options?.map((option) => {
+    let status: 'correct' | 'incorrect' | 'neutral' = 'neutral';
+
+    if (variant === 'result') {
+      if (option.id === currentQuestion.correctOptionId) {
+        status = 'correct';
```
Comment (souzajr) [inline src/components/Quiz/Quiz.tsx:162]: Usar enum

File: src/components/Quiz/Quiz.tsx [hunk 2]
```diff
@@ -117,15 +146,34 @@ const QuizContent = forwardRef<
   );
 });
 
-const QuizAlternative = () => {
+const QuizAlternative = ({
+  variant = 'default',
+}: {
+  variant?: 'result' | 'default';
```
Comment (souzajr) [inline src/components/Quiz/Quiz.tsx:152]: Tipar em um arquivo de types

File: src/components/Quiz/Quiz.tsx [hunk 3]
```diff
@@ -482,12 +531,246 @@ const QuizFooter = forwardRef<
...
+
+  return (
+    <div
+      ref={ref}
+      className={`flex flex-row pt-4 justify-between ${className}`}
+      {...props}
+    >
+      <p className="text-text-950 font-bold text-2xl">Resultado</p>
+      <Badge variant="solid" action="info">
+        {quizType || 'Enem'}
```
Comment (souzajr) [inline src/components/Quiz/Quiz.tsx:551]: Usar enum para os tipos de quiz

File: src/components/Quiz/Quiz.tsx [hunk 4]
```diff
@@ -482,12 +531,246 @@ const QuizFooter = forwardRef<
...
+      getTotalQuestions,
+      timeElapsed,
+      formatTime,
+      bySimulado,
+      byAtividade,
+      byAula,
+    } = useQuizStore();
+
+    const totalQuestions = getTotalQuestions();
+    const quiz = bySimulado || byAtividade || byAula;
```
Comment (souzajr) [inline src/components/Quiz/Quiz.tsx:587]: Usar padrão inglês

File: src/components/Quiz/Quiz.tsx [hunk 5]
```diff
@@ -482,12 +531,246 @@ const QuizFooter = forwardRef<
...
+    if (quiz) {
+      quiz.questions.forEach((question) => {
+        const userAnswer = question.answerKey;
+        const isCorrect = userAnswer && userAnswer === question.correctOptionId;
+
+        if (isCorrect) {
+          correctAnswers++;
+        }
+
+        if (question.difficulty === 'FACIL') {
```
Comment (souzajr) [inline src/components/Quiz/Quiz.tsx:606]: Enum para a dificuldade da questão

File: src/components/Quiz/Quiz.tsx [hunk 6]
```diff
@@ -482,12 +531,246 @@ const QuizFooter = forwardRef<
...
+          totalEasyQuestions++;
+          if (isCorrect) {
+            correctEasyAnswers++;
+          }
+        } else if (question.difficulty === 'MEDIO') {
```
Comment (souzajr) [inline src/components/Quiz/Quiz.tsx:611]: Enum

File: src/components/Quiz/Quiz.tsx [hunk 7]
```diff
@@ -482,12 +531,246 @@ const QuizFooter = forwardRef<
...
+          totalMediumQuestions++;
+          if (isCorrect) {
+            correctMediumAnswers++;
+          }
+        } else if (question.difficulty === 'DIFICIL') {
```
Comment (souzajr) [inline src/components/Quiz/Quiz.tsx:616]: Enum

File: src/components/Quiz/useQuizStore.ts [hunk 8]
```diff
@@ -172,68 +172,74 @@ export const useQuizStore = create<QuizState>()(
...
+          if (!quiz) return;
+
+          // Update the question's answerKey in the quiz
+          const updatedQuestions = quiz.questions.map((question) =>
+            question.id === questionId
+              ? { ...question, answerKey: answerId }
+              : question
           );
+
+          let quizType: 'bySimulado' | 'byAtividade' | 'byAula';
```
Comment (souzajr) [inline src/components/Quiz/useQuizStore.ts:187]: Usar padrão inglês. Tipar em arquivo separado e criar enum

File: src/components/Quiz/useQuizStore.ts [hunk 9]
```diff
@@ -172,68 +172,74 @@ export const useQuizStore = create<QuizState>()(
...
             });
-            // Add to user answers as skipped
-            addUserAnswer(currentQuestion.id);
           }
         },
 
         addUserAnswer: (questionId, answerId) => {
-          const { userAnswers, bySimulado, byAtividade, byAula } = get();
+          const { bySimulado, byAtividade, byAula } = get();
           const quiz = bySimulado || byAtividade || byAula;
```
Comment (souzajr) [inline src/components/Quiz/useQuizStore.ts:218]: Usar inglês

File: src/components/Quiz/useQuizStore.ts [hunk 10]
```diff
@@ -285,25 +291,25 @@ export const useQuizStore = create<QuizState>()(
...
 
         getUnansweredQuestions: () => {
-          const {
-            bySimulado,
-            byAtividade,
-            byAula,
-            selectedAnswers,
-            skippedQuestions,
-          } = get();
+          const { bySimulado, byAtividade, byAula, skippedQuestions } = get();
```
Comment (souzajr) [inline src/components/Quiz/useQuizStore.ts:305]: Usar inglês

File: src/components/Quiz/useQuizStore.ts [hunk 11]
```diff
@@ -1,14 +1,20 @@
...
+  DIFICIL = 'DIFICIL',
+}
+
+export interface Question {
   id: string;
   questionText: string;
   correctOptionId: string;
   description: string;
   type: 'ALTERNATIVA' | 'DISSERTATIVA' | 'MULTIPLA_CHOICE';
   status: 'APROVADO' | 'REPROVADO';
```
Comment (souzajr) [inline src/components/Quiz/useQuizStore.ts:16]: Adicionar também enums aqui. Confere com o Meira os valores que estão sendo usados no backend pra ficar alinhado

File: src/components/Quiz/useQuizStore.ts [hunk 12]
```diff
@@ -1,14 +1,20 @@
 import { create } from 'zustand';
 import { devtools } from 'zustand/middleware';
 
-interface Question {
+export enum Difficulty {
```
Comment (souzajr) [inline src/components/Quiz/useQuizStore.ts:4]: Vamos manter o padrão do sistema, algo como QUESTION_DIFFICULTY

**Review by souzajr — CHANGES_REQUESTED** [review]

**Review by souzajr — APPROVED** [review]
